ui/src/app/utils/user-apps-utils.tsx

```diff
@@ -0,0 +1,69 @@
+import { AppStatus, CreateAppRequest } from 'generated/fetch';
+
+import { leoAppsApi } from 'app/services/notebooks-swagger-fetch-clients';
+import { appsApi } from 'app/services/swagger-fetch-clients';
+
+import { userAppsStore } from './stores';
+
+const appStatusesRequiringUpdates = [
+  AppStatus.DELETING,
+  AppStatus.PROVISIONING,
+  AppStatus.STARTING,
+  AppStatus.STOPPING,
+];
+
+const doUserAppsRequireUpdates = () => {
+  const { userApps } = userAppsStore.get();
+  return (
+    !userApps ||
+    userApps
+      .map((userApp) => userApp.status)
+      .some((appStatus) => appStatusesRequiringUpdates.includes(appStatus))
+  );
+};
+
+export const maybeStartPollingForUserApps = (namespace) => {
+  const { updating } = userAppsStore.get();
+  // Prevents multiple update processes from running concurrently.
+  if (updating) {
+    return;
+  }
+
+  userAppsStore.set({ ...userAppsStore.get(), updating: true });
+  appsApi()
+    .listAppsInWorkspace(namespace)
+    .then((listAppsResponse) => {
+      userAppsStore.set({ userApps: listAppsResponse, updating: false });
+      if (doUserAppsRequireUpdates()) {
+        setTimeout(() => {
+          maybeStartPollingForUserApps(namespace);
+        }, 10 * 1000);
+      }
+    });
+};
+
+export const createUserApp = (namespace, config: CreateAppRequest) => {
+  return appsApi()
+    .createApp(namespace, config)
+    .then(() => {
+      maybeStartPollingForUserApps(namespace);
+    });
+};
```

Comment:
Love how clean and clear these are. I prefer the syntax that omits the block whenever I don't need it because it calls to attention the cases where I do need it.
```suggestion
export const createUserApp = (namespace, config: CreateAppRequest) =>
  appsApi()
    .createApp(namespace, config)
    .then(() => maybeStartPollingForUserApps(namespace));
```

Reply:
I like the look of this too. Thanks for pointing it out.